src/emu/render.cpp

```diff
@@ -1574,38 +1574,80 @@ void render_target::update_layer_config()
 
 void render_target::load_layout_files(const internal_layout *layoutfile, bool singlefile)
 {
-	bool have_default = false;
+	bool have_default  = false;
+	bool have_artwork  = false;
+	bool have_override = false;
 
 	// if there's an explicit file, load that first
 	const char *basename = m_manager.machine().basename();
 	if (layoutfile)
-		have_default |= load_layout_file(basename, layoutfile);
+		have_artwork |= load_layout_file(basename, layoutfile);
 
 	// if we're only loading this file, we know our final result
 	if (singlefile)
 		return;
 
-	// try to load a file based on the driver name
+	// if override_artwork defined, load that and skip artwork other than default
+	if (m_manager.machine().options().override_artwork())
+	{
+		if (load_layout_file(m_manager.machine().options().override_artwork(), m_manager.machine().options().override_artwork()))
+			have_override |= true;
```

Comment:
What's the point of `|= true` on a `bool`?  It forces it to true, so you may as well just do `= true` (same thing happens again below).

Reply:
None whatsoever, but since the original (current) code used the same construction I figured I'd follow that coding style. If you prefer I can change that throughout this function.

Reply:
There are several things to think about.  Firstly, using `|=` with C++ `bool` is kind of weird.  This statement:
```
some_bool |= other_bool;
```
actually decays to something like this:
```
some_bool = ((some_bool ? 1 : 0) | (other_bool ? 1 : 0)) != 0;
```
Of course, the optimiser should be able to do a decent job of handling it, but it's still a case of using `bool` in `int` context and several implicit conversions.

Consider this:
```
some_bool |= function_returning_bool();
```
In this case you can't replace `|=` with an assignment here because if `some_bool` was already `true` but the function returns `false`, you still want `some_bool` to be `true` afterwards.  You could write it like this if you want to avoid implicit conversions:
```
some_bool = function_returning_bool() || some_bool;
```
However, that's a bit less concise (note that the order can't be reversed around the short-circuit operator `||` or it won't call the function if `some_bool` is already `true`).

But when you consider this:
```
some_bool |= true;
```
The initial value of `some_bool` is completely irrelevant - it's effectively assigning it `true` but in a more confusing way.  I'd say this kind of thing should be eliminated, including the two that already exist in `src/emu/render.cpp` (as of 0f413001dc0034412779747fde5b003bcfecc2f0).